src/coordination/include/coordination/coordination_observer.hpp

```diff
@@ -0,0 +1,33 @@
+// Copyright 2024 Memgraph Ltd.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt; by using this file, you agree to be bound by the terms of the Business Source
+// License, and you may not use this file except in compliance with the Business Source License.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+#ifdef MG_ENTERPRISE
+
+#pragma once
+
+#include "coordination/coordinator_communication_config.hpp"
+
+namespace memgraph::coordination {
+
+class CoordinatorInstance;
+
+class CoordinationClusterChangeObserver {
+ public:
+  explicit CoordinationClusterChangeObserver(CoordinatorInstance *instance);
+  void Update(std::vector<CoordinatorToCoordinatorConfig> const &configs);
```

Comment:
this could be just lambda function for less-instrusive Observer pattern. No need for whole class, just lambda function which captures pointer to CoordinatorInstance, then you don't need new files etc. you just pass functions

Reply:
I understand. Having class means better readability and separation. Usually lambas are used for strategy pattern. I would leave it like this rather if you don't mind

Reply:
Well the question is why is this more readable and why separation? There is some new object (Observer) but what do we get by having it as a class, not much IMO just few more files. Lambdas are often used in such situations also in Observer not just in Strategy.

I will leave this one to you to decide then

Reply:
Readable as we don't create a lambda function in the constructor of the coordinator instance, so we don't pollute the constructor and the coordinator instance is in that way the owner of the function with lambda. On the other hand this way, the coordinator instance via some other class is connected to the state manager, which sounds a lot like separation of concerns.
 So I would leave it like this.